presto-main/src/main/java/io/prestosql/server/protocol/Query.java

```diff
@@ -702,6 +702,7 @@ private static QueryError toQueryError(QueryInfo queryInfo)
                 null,
                 errorCode.getCode(),
                 errorCode.getName(),
+                queryInfo.getFailureInfo().getSemanticErrorCode().name(),
```

Comment:
This has a similar bug for `.getFailureInfo()` returning null, which we check for above. It might only be possible if there is a bug elsewhere, but we should handle it anyway, to be consistent with the check above.

Let's change the block above (starting line 683) to look like
```java
ExecutionFailureInfo executionFailure;
if (queryInfo.getFailureInfo() != null) {
    executionFailure = queryInfo.getFailureInfo();
}
else {
    log.warn("Query %s in state %s has no failure info", queryInfo.getQueryId(), state);
    executionFailure = toFailure(new RuntimeException(format("Query is %s (reason unknown)", state)));
}
FailureInfo failure = executionFailure.toFailureInfo();
```
Then we can use `executionFailure` here without the risk of it being null.